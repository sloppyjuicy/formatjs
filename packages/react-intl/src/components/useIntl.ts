import * as React from 'react'
import {IntlShape} from '../types'
import {invariantIntlContext} from '../utils'
import {Context} from './injectIntl'

export default function useIntl(this: void): IntlShape {
  const intl = React.useContext(Context)
  invariantIntlContext(intl)
  return intl
}
