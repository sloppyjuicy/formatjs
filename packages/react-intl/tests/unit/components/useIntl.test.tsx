import {render} from '@testing-library/react'
import * as React from 'react'
import {IntlProvider} from '../../..'
import useIntl from '../../../src/components/useIntl'

const FunctionComponent = ({spy}: {spy?: Function}) => {
  const hookReturns = useIntl()
  spy!(hookReturns.locale)
  return null
}

const FC = () => {
  const {formatNumber} = useIntl()
  return formatNumber(10000, {style: 'currency', currency: 'USD'}) as any
}

describe('useIntl() hook', () => {
  it('throws when <IntlProvider> is missing from ancestry', () => {
    // So it doesn't spam the console
    jest.spyOn(console, 'error').mockImplementation(() => {})
    expect(() => render(<FunctionComponent />)).toThrow(
      '[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.'
    )
  })

  it('hooks onto the intl context', () => {
    const spy = jest.fn()
    render(
      <IntlProvider locale="en">
        <FunctionComponent spy={spy} />
      </IntlProvider>
    )

    expect(spy).toHaveBeenCalledWith('en')
  })

  it('should work when switching locale on provider', () => {
    const {rerender, getByTestId} = render(
      <IntlProvider locale="en">
        <span data-testid="comp">
          <FC />
        </span>
      </IntlProvider>
    )
    expect(getByTestId('comp')).toMatchSnapshot()
    rerender(
      <IntlProvider locale="es">
        <span data-testid="comp">
          <FC />
        </span>
      </IntlProvider>
    )
    expect(getByTestId('comp')).toMatchSnapshot()
    rerender(
      <IntlProvider locale="en">
        <span data-testid="comp">
          <FC />
        </span>
      </IntlProvider>
    )

    expect(getByTestId('comp')).toMatchSnapshot()
  })
})
