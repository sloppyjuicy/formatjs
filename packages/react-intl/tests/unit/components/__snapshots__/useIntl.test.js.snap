// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useIntl() hook should work when switching locale on provider 1`] = `
<span
  data-testid="comp"
>
  $10,000.00
</span>
`;

exports[`useIntl() hook should work when switching locale on provider 2`] = `
<span
  data-testid="comp"
>
  10.000,00Â US$
</span>
`;

exports[`useIntl() hook should work when switching locale on provider 3`] = `
<span
  data-testid="comp"
>
  $10,000.00
</span>
`;
